src/Microsoft.Dnx.Compilation.Abstractions/CompilationProjectContext.cs

```diff
@@ -24,6 +27,9 @@ public class CompilationProjectContext
             CompilationTarget target,
```

Comment:
Personally I find a long constructor very discouraging to use. You need to remember the sequence of the parameter or heavily relies on the IDE to give you a clue. I suggest to make properties settable.